### PR TITLE
add a Sequelize option

### LINKED ISSUE #20: Reference to Sequelize
It's possible pass the reference to Sequelize in the sequelize storage constructor instead of doing this?

[https://github.com/sequelize/umzug/blob/master/lib/storages/sequelize.js#L8](https://github.com/sequelize/umzug/blob/master/lib/storages/sequelize.js#L8)
```js
var Sequelize = require(resolve('sequelize', { basedir: process.cwd() }));
```

I'm working on a framework that depends on this library, but the applications using the framework doesn't depends directly on sequelize.

[https://github.com/axelhzf/barbakoa/blob/master/app/server/migrations.js](https://github.com/axelhzf/barbakoa/blob/master/app/server/migrations.js)

Does this make sense?

### PR DESCRIPTION
**Pull request by axelhzf**
fixes #20

### CODE REVIEW

File: lib/storages/sequelize.js [hunk 1]
```diff
@@ -43,11 +42,23 @@ module.exports = redefine.Class({
...
     }, this.options.storageOptions || {});
-
+    
+    if(!this.options.storageOptions.columnType) {
+      // You may need to override this for utf8mb4 charsets
+      var Sequelize;
+      try {
+        Sequelize = require(resolve.sync('sequelize', { basedir: process.cwd() }))
+      } catch(e) {
+        Sequelize = this.options.storageOptions.Sequelize;
```
Comment (sdepold) [inline lib/storages/sequelize.js]: I wonder if we should check for this option first?

**Comment by sdepold** [comment]
One thing that came into my mind tonight: Couldn't we just `sequelize.constructor` ?

**Comment by axelhzf** [comment]
Yes, to use `sequelize.constructor` is a cleaner solution. I have modified the PR

File: lib/storages/sequelize.js [hunk 2]
```diff
@@ -4,8 +4,7 @@ var _         = require('lodash');
 var Bluebird  = require('bluebird');
 var path      = require('path');
 var redefine  = require('redefine');
-var resolve   = require('resolve').sync;
-var Sequelize = require(resolve('sequelize', { basedir: process.cwd() }));
+var resolve   = require('resolve');
```
Comment (sdepold) [inline lib/storages/sequelize.js]: I think we don't need this anymore?

File: lib/storages/sequelize.js [hunk 3]
```diff
@@ -56,6 +53,7 @@ module.exports = redefine.Class({
     if (!this.options.storageOptions.model) {
       var sequelize = this.options.storageOptions.sequelize;
       var modelName = this.options.storageOptions.modelName;
+      var columnType = this.options.storageOptions.columnType || sequelize.constructor.STRING;
```
Comment (sdepold) [inline lib/storages/sequelize.js]: Could you do me a favor and add a line `var Sequelize = sequelize.constructor` ?

File: lib/storages/sequelize.js (hunk 2 above)
Comment (axelhzf) [inline lib/storages/sequelize.js]: yes, you're right

File: lib/storages/sequelize.js (hunk 3 above)
Comment (axelhzf) [inline lib/storages/sequelize.js]: Yes, I just push a new commit with the two changes

**Comment by sdepold** [comment]
Awesome :)

**Pull request merged**